app/models/krikri/validation_report.rb

```diff
@@ -0,0 +1,54 @@
+module Krikri
+  class ValidationReport
+    attr_accessor :provider_id, :start, :rows
+
+    REQUIRED_FIELDS = ['dataProvider_name', 'isShownAt_id', 'preview_id',
+                       'sourceResource_rights', 'sourceResource_title',
+                       'sourceResource_type_id']
+
+    ##
+    # @param &block may contain provider_id
+    #   Sample use:
+    #     ValidationReport.new.all do
+    #       self.provider_id = '0123'
+    #     end
+    #
+    # @return [Array<Blacklight::SolrResponse::Facet>]
+    def all(&block)
+      # set values from block
+      instance_eval &block if block_given?
+
+      query_params = { :rows => 0,
+                       'facet.field' => REQUIRED_FIELDS,
+                       'facet.mincount' => 10000000,
+                       'facet.missing' => true }
+      query_params[:fq] = "provider_id:\"#{@provider_id}\"" if
+        @provider_id.present?
```

Comment:
Hold the phones! I just realized that when I load `http://localhost:3000/krikri/validation_reports/sourceResource_title?provider=123` into my browser, I should not see any records, because the record in the test that appears on the `sourceResource_title` validation report has a `provider_id` of `_:b3`.  So things in the test data ARE working as expected.

However, _if_ there were test data in which a record with a valid `provider_id` had, say, a missing value for `sourceResource_title`, you may still run into trouble with that local name, as described above.  It would be worth testing to find out.